### ISSUE TITLE
custom-error-definition still has false positive on TypeScript class property

### ISSUE THREAD

**Issue by felixfbecker** [sindresorhus/eslint-plugin-unicorn#753]
```ts
const REV_NOT_FOUND_ERROR_NAME = 'RevNotFoundError' as const
export class RevNotFoundError extends Error {
    public readonly name = REV_NOT_FOUND_ERROR_NAME
    constructor(rev?: string) {
        super(`rev ${String(rev)} not found`)
    }
}
```
triggers an error from `custom-error-definition`, but should not.

**Comment by fisker** [comment]
This rule has known issues. I'm rewriting it https://github.com/sindresorhus/eslint-plugin-unicorn/pull/680 , but don't have enough time to finish it.

**Comment by fregante** [comment]
Closing in favor of https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1993
